### PR TITLE
284 Render helper ensuredir function bug

### LINKED ISSUE #284: Error in render_helper.php file occurring in render_template func via ensure_dir func
In render_helper.php file in function error was identified: Ouch!! It seems that the /var/www/wp-content/themes/<>/tmp

ensure_dir function sometimes cannot override the tmp folder permission to writeable, some server forbid the chmod action. Hence we compile our files via CLI during CI build process, this generates files in tmp DIR. Now I have corrected version of ensure_dir function test in our environment to check both compiled files in tmp and also create tmp if didn't exist this applies for both scenario.

Secondly we have filter 'wordless_environment' to override the environment.

PR: https://github.com/welaika/wordless/pull/283

### PR DESCRIPTION
**Pull request by alessandro-fazzi**
Fix #284 

Previously proposed by @arun744-newscorp, but never brought to the end. I've just added the needed type declarations to the brought-back original #283 

:)

### CODE REVIEW

File: wordless/helpers/render_helper.php [hunk 1]
```diff
@@ -246,21 +248,30 @@ function wl_yield() {
...
+
+        $dir_exists_and_writable = false;
 
         if (!file_exists($dir)) {
             mkdir($dir, 0770);
         }
 
         if (!is_writable($dir)) {
-            chmod($dir, 0770);
+            chmod($dir, 770);
```
Comment (danielegiorgis) [inline wordless/helpers/render_helper.php:267]: mmm -_-.. maybe my ignorance... is it the same thing compared to 0770?

Comment (alessandro-fazzi) [reply wordless/helpers/render_helper.php:267]: Don't know: I'm not the author. Will check the docs.

**Comment by alessandro-fazzi** [comment]
Rebased on master

File: wordless/helpers/render_helper.php (hunk 1 above)
Comment (alessandro-fazzi) [reply wordless/helpers/render_helper.php]: Permission uniformed to use 4 digits expression. TY

**Review by danielegiorgis — APPROVED** [review]

**Pull request merged**